### PR TITLE
./please tools signin

### PR DESCRIPTION
**Pull request by La0**

### CODE REVIEW

File: lib/please_cli/please_cli/local_config.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+
+log = cli_common.log.get_logger(__name__)
+
+
+class LocalConfig(object):
+    '''
+    Config for developers, in their home folder
+    '''
+    def __init__(self):
+        self.path = os.path.expanduser('~/.pleaserc')
```
Comment (garbas) [inline lib/please_cli/please_cli/local_config.py]: Lets use [XDG Base Directory Specification](https://specifications.freedesktop.org/basedir-spec/basedir-spec-latest.html) for the location of the user configuration.

We should read from `XDG_CONFIG_HOME` system variable and default to `~/.config`. And for future lets us also create a folder where we put please configuration. User configuration should then be in `~/.config/please/config.toml`

File: lib/please_cli/please_cli/config.py [hunk 2]
```diff
@@ -65,6 +67,8 @@
 PROJECTS += ['scriptworker/shipitscript']
 PROJECTS += DEV_PROJECTS
 
+LOCAL = LocalConfig()
```
Comment (garbas) [inline lib/please_cli/please_cli/config.py]: I think LOCAL is not really a descriptive name maybe:
```suggestion
CONFIG = LocalConfig()
```

File: please [hunk 3]
```diff
@@ -32,7 +32,7 @@ if uname -a | grep -q Darwin; then
 fi
 
 if [ "$USE_NIX" = "1" ]; then
-  drv=$(nix-instantiate nix/default.nix -A please-cli) 2> /dev/null
+  drv=$(nix-instantiate nix/default.nix -A please-cli)
```
[the remark this answers is not in the document]
Comment (La0) [reply please]: Oops, i did not mean to commit that :cry:

File: lib/please_cli/please_cli/local_config.py (hunk 1 above)
Comment (La0) [reply lib/please_cli/please_cli/local_config.py]: Ok :+1:

File: please [hunk 4]
```diff
@@ -58,6 +60,7 @@ else
             --name "$DOCKER_NAME" \
             --volume="$(pwd)":/app \
             --volume=/var/run/docker.sock:/var/run/docker.sock \
+            --volume=$USER_FOLDER:/home/app/.config/please:rw \
```
Comment (La0) [inline please]: The persistence will only work after rebuilding the base image, or you'll get a permission denied

File: lib/please_cli/please_cli/signin.py [hunk 5]
```diff
@@ -0,0 +1,83 @@
...
+    # Wait for credentials
+    click.echo('Waiting for HTTP request...')
+    while not httpd.taskcluster_credentials:
+        httpd.handle_request()
+
+    # Write credentials
+    config.write_user_config({
+        'taskcluster': httpd.taskcluster_credentials,
+    })
+    click.secho('Tasklcluster credentials saved !', fg='green')
```
Comment (garbas) [inline lib/please_cli/please_cli/signin.py]:
```suggestion
    click.secho('Taskcluster credentials saved !', fg='green')
```

File: lib/please_cli/please_cli/signin.py [hunk 6]
```diff
@@ -0,0 +1,83 @@
...
+    click.secho('Please use this url: {}'.format(client_url), fg='blue')
+
+    # Wait for credentials
+    click.echo('Waiting for HTTP request...')
+    while not httpd.taskcluster_credentials:
+        httpd.handle_request()
+
+    # Write credentials
+    config.write_user_config({
+        'taskcluster': httpd.taskcluster_credentials,
```
Comment (garbas) [inline lib/please_cli/please_cli/signin.py]: I think the structure should be different.  it should end up with:
```toml
[common]
taskcluster_client_id = ...
taskcluster_access_token = ...
```

**Pull request merged**